### PR TITLE
Gdr 2028

### PR DESCRIPTION
**Pull request by bczech**
# Description
## What changed?
Switch from merge to [[
Related JIRA issue: GDR-2028

## Why was it changed?
To improve performance

# Checklist for sustainable code base
- [ ] I added tests for any code changed/added
- [ ] I added documentation for any code changed/added
- [ ] I made sure naming of any new functions is self-explanatory and consistent

# Logistic checklist
- [X] Package version bumped
- [X] Changelog updated

# Screenshots (optional)

### CODE REVIEW

**Review by gladkia — APPROVED** [review]

File: R/convert_mae_se_assay_to_dt.R [hunk 1]
```diff
@@ -80,16 +79,16 @@ convert_se_assay_to_dt <- function(se,
...
-  cData <- colData(se)
-  cData$cId <- rownames(cData)
-  ids <- expand.grid(rData$rId, cData$cId)
-  colnames(ids) <- c("rId", "cId")
-  ids[] <- lapply(ids, as.character)
-  annotations <- merge(ids, rData, by = "rId", all.x = TRUE)
-  annotations <- merge(annotations, cData, by = "cId", all.x = TRUE)
-  merge(dt, annotations, by = c("rId", "cId"), all.x = TRUE)
+  rData <- data.table::as.data.table(rowData(se))
+  rData[, rId := rownames(se)]
```
Comment (DominikZabinski) [inline R/convert_mae_se_assay_to_dt.R:83]:
```suggestion
  rData$rId <- rownames(se)
```

File: R/convert_mae_se_assay_to_dt.R [hunk 2]
```diff
@@ -80,16 +79,16 @@ convert_se_assay_to_dt <- function(se,
...
+  cData <- data.table::as.data.table(colData(se))
+  cData[, cId := colnames(se)]
```
Comment (DominikZabinski) [inline R/convert_mae_se_assay_to_dt.R:85]:
```suggestion
  cData$cId <- colnames(se)
```

File: R/helpers-tests.R [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    "E0" = values,
+    "EC50" = values,
+    "h RV" = values,
+    "GR50" = values,
+    "IC50" = values,
+    "GR Max" = values,
+    "E Max" = values,
+    "x" = values,
+    "Concentration" = values,
+    check.names = FALSE
```
Comment (DominikZabinski) [inline R/helpers-tests.R:81]: `data.table::data.table` has `check.names = FALSE` by default.

File: R/helpers-tests.R [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+#' Function for generating local synthetic data used for unit tests in modules
+#'
+#' @examples 
+#' gen_synthetic_data()
+#'
+#' @return list with drugs, cell_lines, raw_data and assay_data
+#'
+#' @export
+gen_synthetic_data <- function(m, n) {
+  drug_names <- paste0("drug_00", 1:m)
```
Comment (DominikZabinski) [inline R/helpers-tests.R:58]: Could we add simple checks like `checkmate::assert_integer(m, lower = 1)`?

**Review by DominikZabinski — APPROVED** [review]

File: R/helpers-tests.R [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+#'
+#' Function for generating local synthetic data used for unit tests in modules
+#'
+#' @examples 
+#' gen_synthetic_data()
+#'
+#' @return list with drugs, cell_lines, raw_data and assay_data
+#'
+#' @export
+gen_synthetic_data <- function(m, n) {
```
Comment (DominikZabinski) [inline R/helpers-tests.R:53]: Please provide default values for those parameters

**Review by DominikZabinski — APPROVED** [review]

**Pull request merged**